src/agent/src/datasource.ts

```diff
@@ -17,10 +18,16 @@ export class Datasource {
   }
 
   static async Loki(host: string) {
+    if (this.cache !== undefined) {
+      return this.cache;
+    }
+
     this.datasource ??= new Datasource(host);
 
     const datasources = await this.datasource.fetchDatasources();
+    const datasource = datasources.find((datasource) => datasource.type === "loki" && datasource.isDefault)!;
```

Comment:
Could be a problem if there is no isDefault right?

```js
const lokiDatasources = datasources.filter((datasource) => datasource.type === "loki");
// Throw if lokiDatasources.length is 0?

const datasource = lokiDatasources.find((datasource) => datasource.isDefault) ?? lokiDatasources.at(0)!;
```

Reply:
Indeed

I don't think we should throw if `datasources.length` is 0, it should not be possible for me. We are sending an alert so the Loki API is OK so we have a Loki datasource. WDYT ?